src/utils.c

```diff
@@ -455,6 +455,18 @@ SEXP new_empty_ordered(SEXP levels) {
   return out;
 }
 
+// [[ include("utils.h") ]]
+bool list_has_inner_names(SEXP x, R_len_t size) {
+  for (R_len_t i = 0; i < size; ++i) {
+    SEXP elt = VECTOR_ELT(x, i);
+    if (vec_names(elt) != R_NilValue) {
```

Comment:
```suggestion
    if (r_names(elt) != R_NilValue) {
```

Reply:
Is this right? For matrices we do want the row names, and `r_names()` would return `NULL` I think

Reply:
oops you're right! Maybe rename to `list_has_inner_vec_names()`?

By the way should we combine outer and inner vec-names? It doesn't seem right to ignore outer names if they exist.

Reply:
I chose to ignore them since the output from `vec_chop()` would never generate a named list and I figured we'd generally be inverting that or something like it. But I can look into using a `name_spec` argument like `vec_c()` if you think that makes sense?

Reply:
I think it makes sense because the caller might have added names to keep track of some metadata.